openfecwebapp/data_mappings.py

```diff
@@ -0,0 +1,195 @@
+            value_map[v], grouping=True)
+        else:
+            totals_mapped[v] = 'unavailable'
+
+    if reports.get('report_year'):
+        totals_mapped['report_year'] = str(int(reports['report_year']))
+
+    # "calculated from" on site
+    if reports.get('election_cycle'):
+        cycle_minus_one = str(int(reports['election_cycle']) - 1)
+        totals_mapped['years_totals'] = cycle_minus_one 
+        totals_mapped['years_totals'] += ' - ' 
+        totals_mapped['years_totals'] += str(reports['election_cycle'])
+
+    # "source:" on site
+    if reports.get('report_type_full'):
+        totals_mapped['report_desc'] = re.sub('{.+}', 
+            '', reports['report_type_full']) 
+
+    return totals_mapped
+
+# we want to show the committees on their related candidate 
+# pages in this order, with primary committees on top
+committee_type_map = {
+    'A': 'authorized_committees',
+    'D': 'leadership_committees',
+    'J': 'joint_committees'
+}
+
+def map_candidate_page_values(c):
+    candidate = map_candidate_table_values(c)
+
+    if c.get('elections'):
+        c_e = c['elections'][0]
+        candidate['incumbent_challenge'] = c_e.get(
+            'incumbent_challenge_full', '')
+        if c_e.get('primary_committee'):
+            candidate['primary_committee'] = _map_committee_values(
+                c_e['primary_committee'])
+            candidate['related_committees'] = True
+
+        # affiliated committees = committee_type_map, 
+        # plus more we ignore for the candidate pages
+        if c_e.get('affiliated_committees'):
+            candidate['affiliated_committees'] = []
+            for i in range(len(c_e['affiliated_committees'])):
+                candidate['affiliated_committees'].append(
+                    _map_committee_values(c_e[
+                        'affiliated_committees'][i]))
+
+            candidate['authorized_committees'] = []
+            candidate['leadership_committees'] = []
+            candidate['joint_committees'] = []
+
+            for i in range(len(candidate['affiliated_committees'])):
+                cmte = candidate['affiliated_committees'][i]
+                cmte_type = cmte['designation_code']
+                # drop anything that's not of the types we're
+                # interested in
+                if (cmte_type in committee_type_map):
+                    candidate[committee_type_map[
+                        cmte_type]].append(cmte)
+                    candidate['related_committees'] = True
+
+    return candidate
+
+def map_committee_page_values(c):
+    committee = map_committee_table_values(c)
+
+    if c.get('address'):
+        committee['address'] = {
+            'state': c['address'].get('state'),
+            'zip': c['address'].get('zip'),
+            'city': c['address'].get('city'),
+            'street_1': c['address'].get('street_1'),
+            'street_2': c['address'].get('street_2')
+        }
```

Comment:
Hah. Hahahaha. Hahahahahahaha. I was so in the mode of.... ok. This never happened.